packages/stokhos/test/UnitTest/Stokhos_SacadoMPVectorUnitTest.cpp

```diff
@@ -474,52 +516,73 @@ struct UnitTestSetup {
                           setup.rtol, setup.atol, out);                 \
   }
 
-namespace DynamicVecTest {
-  typedef Kokkos::DefaultExecutionSpace execution_space;
-  typedef Stokhos::DynamicStorage<int,double,execution_space> storage_type;
-  typedef Sacado::MP::Vector<storage_type> vec_type;
-  typedef UnitTestSetup<vec_type> UTS;
-  VECTOR_UNIT_TESTS(DynamicVector)
-}
+/**
+ * Series of tests to run for complex type.
+ * It will run the series of tests for any type.
+ */
+#define VECTOR_UNIT_TESTS_COMPLEX_TYPE(VEC,SCALAR_T)                               \
+  /* Run the series of tests for any type. */                                      \
+  VECTOR_UNIT_TESTS_ANY_TYPE(VEC,SCALAR_T)
 
-namespace DynamicStridedVecTest {
-  typedef Kokkos::DefaultExecutionSpace execution_space;
-  typedef Stokhos::DynamicStridedStorage<int,double,execution_space> storage_type;
-  typedef Sacado::MP::Vector<storage_type> vec_type;
-  typedef UnitTestSetup<vec_type> UTS;
-  VECTOR_UNIT_TESTS(DynamicStridedVector)
-}
+/**
+ * Series of tests to run for real type.
+ * It will run the series of tests for any type, as well as tests that don't work or
+ * don't make sense for complex type.
+ */
+#define VECTOR_UNIT_TESTS_REAL_TYPE(VEC,SCALAR_T)                                  \
+  /* Run the series of tests for any type. */                                      \
+  VECTOR_UNIT_TESTS_ANY_TYPE(VEC,SCALAR_T)                                         \
+  /* Operator cbrt not supported for complex type but supported for real type. */  \
+  UNARY_UNIT_TEST(VEC, SCALAR_T, cbrt , Cbrt, using std::cbrt;)                    \
+  /* Operator atan2 not supported for complex type but supported for real type. */ \
+  BINARYFUNC_UNIT_TEST(VEC, SCALAR_T, atan2, atan2, using std::atan2;, ATan2)      \
+  /* Operators min and max don't work for complex type because UnitTestSetup hard-codes the value of 'a' as a double. */ \
+  BINARYFUNC_UNIT_TEST(VEC, SCALAR_T, max  , max  , using std::max  ;, Max)        \
+  BINARYFUNC_UNIT_TEST(VEC, SCALAR_T, min  , min  , using std::min  ;, Min)        \
```

Comment:
In `Kokkos_ArithTraits.hpp`, there are `min` and `max` functions for complex types.  Could be interesting to point to those?

Reply:
If I remember well, we discussed the use of Kokkos arithmetic traits in the past but decided not to use it because some specializations had been commented out.
Why would we use it for `max`?

Reply:
My comment was indeed not very clear.  Not sure what the best point of view is here.  One point of view could be to decide whether there are `min` and `max` functions, indeed not necessarily those from ArithTraits, that are defined sample wise and that we want to extend to ensembles and then also test the extension.   Do calls with `min` and `max` and/or 'std::min' and `std::max` work sample wise?

Other than that, the comment just above these lines says that `min` and `max` are not tested because `a` is hardcoded as a double in the test setup.  Wouldn't there be a conversion to complex type if `a` is assigned e.g. to `cx` next.

Reply:
> Other than that, the comment just above these lines says that min and max are not tested because a is hardcoded as a double in the test setup. Wouldn't there be a conversion to complex type if a is assigned e.g. to cx next.µ

I think my comment was wrong. `min` and `max` are simply not supported for complex types :smile:

Reply:
I agree with @etphipp that we cannot define a generic `max`/`min` operator for complex.